style/typescript/.eslintrc.json

```diff
@@ -0,0 +1,48 @@
+{
+  "extends": ["@thoughtbot/eslint-config", "plugin:@typescript-eslint/recommended", "plugin:jest/recommended"],
```

Comment:
I typically prefer to use [eslint-plugin-prettier](https://github.com/prettier/eslint-plugin-prettier) along with [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier) to turn off ESLint rules that overlap with Prettier. Curious to hear your thoughts. We're doing this in the [React Reference app](https://github.com/thoughtbot/react-reference-app/blob/master/.eslintrc.json#L8).

Reply:
I changed this configuration quite a bit in [the .eslintrc.json for react-suspenders](https://github.com/thoughtbot/react-suspenders/pull/1/files#diff-838f0d5f2e5e1cba5ea6e331d430bd94) - I'd love your thoughts on the file in that PR.

My plan is to delete the configs I added to this repo, and instead link into `react-suspenders`.